code/datums/status_effects/debuffs.dm

```diff
@@ -57,7 +65,7 @@
 
 /datum/status_effect/incapacitating/sleeping/tick()
 	if(owner.staminaloss)
-		owner.adjustStaminaLoss(-0.35) //reduce stamina loss by 0.35 per tick, 7 per 2 seconds
```

Comment:
Sleeping takes more than 12.5 seconds anyway so by the time you wake up all the stamina damage is gone.